django/db/models/base.py

```diff
@@ -442,7 +442,10 @@ def __init__(self, *args, **kwargs):
                 if val is _DEFERRED:
                     continue
                 _setattr(self, field.attname, val)
-                kwargs.pop(field.name, None)
+                ignored_kwarg = kwargs.pop(field.name, None)
+                if ignored_kwarg:
+                    raise TypeError(f'{cls.__name__}() got both positional and '
+                                    f"keyword arguments for field '{field.name}'")
```

Comment:
A few observations:
- There is no need for the temporary variable.
- We cannot just check if falsy as the value could be `0` or `False`.
- Compare to a sentinel object to ensure that `None` isn't passed. (I think we can re-use `NOT_PROVIDED` here.)
- Add a period to the end of the exception message.

```suggestion
                if kwargs.pop(field.name, NOT_PROVIDED) is not NOT_PROVIDED:
                    raise TypeError(
                        f'{cls.__name__}() got both positional and keyword '
                        f"arguments for field '{field.name}'."
                    )
```

Reply:
Ouch, good catch on 0/False. Thanks for giving this a look.

Reply:
Is it worth adding a second test with a Falsy value, e.g. `headline=''`?

Reply:
Probably not a bad idea. You could also add, e.g. `views = IntegerField()`, to the `Article` model to test `0`, and other fields to test `False` and `None`

Reply:
I don't think it's worth to add multiple fields. `IntegerField(null=True)` should be enough.

Reply:
Sure, so two extra fields - one to handle `0` and `None`, the other to handle `False`.

Reply:
I think it's enough to test `None` and `''`.